File: internal/indexer/testdata/go/defs_a.go
package sample

import "fmt"

// Config holds application configuration.
// It supports multiple environments.
type Config struct {
	Host     string
	Port     int
	LogLevel string
}

// Validate checks if the config is valid.
func (c *Config) Validate() error {
	if c.Host == "" {
		return fmt.Errorf("host is required")
	}
	if c.Port <= 0 {
		return fmt.Errorf("port must be positive")
	}
	return nil
}

// Formatter formats output.
type Formatter interface {
	Format(data string) string
	Reset()
}

// DefaultPort is the fallback port.
const DefaultPort = 8080

// AppName is the application name.
var AppName = "myapp"

// NewConfig creates a Config with defaults.
func NewConfig(host string) *Config {
	return &Config{
		Host:     host,
		Port:     DefaultPort,
		LogLevel: "info",
	}
}

type (
	// Endpoint describes an API endpoint.
	Endpoint struct {
		Path   string
		Method string
	}

	// Handler processes requests.
	Handler func(req string) string
)

File: internal/indexer/testdata/python/defs_a.py
from typing import Optional

# Default port for the application.
DEFAULT_PORT = 8080

# Application name.
app_name = "myapp"


# AppConfig holds application configuration.
class AppConfig:
    def __init__(self, host: str, port: int = DEFAULT_PORT):
        self.host = host
        self.port = port

    def validate(self) -> bool:
        if not self.host:
            raise ValueError("host is required")
        if self.port <= 0:
            raise ValueError("port must be positive")
        return True

    @property
    def address(self) -> str:
        return f"{self.host}:{self.port}"


class AppFormatter:
    """Formatter for output formatting."""

    def format(self, data: str) -> str:
        return f"[{data}]"

    def reset(self):
        pass


# say_hello is a standalone function.
def say_hello(name: str) -> str:
    greeting = f"Hello, {name}!"
    print(greeting)
    return greeting
